app/models/user.rb

```diff
@@ -0,0 +1,41 @@
+class User < ApplicationRecord
+  has_many :votes
+  validates :username, presence: true, uniqueness: true
+
+
+  # create a vote object with current user's id and the work id passed in
+  def vote(work_id)
+    vote = Vote.where(user_id: self.id, work_id: work_id)
+    return vote
+  end
+
+  # Assign the list of votes to the variable votes_array...helper method
+  def users_votes
+    votes_array = Vote.where(user_id: self.id)
+    return votes_array
+  end
+
+  # collect the work ids for all the works that the user upvoted
+  def upvoted
+    votes = users_votes
+    upvoted_workids = []
+    votes.each do |vote|
+      upvoted_workids << vote
```

Comment:
This loop looks through an array of votes and... for each vote... it shovels the vote into another array named `upvoted_workids`. It doesn't shovel in the vote's work id